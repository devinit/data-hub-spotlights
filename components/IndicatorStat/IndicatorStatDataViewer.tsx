import React, { FunctionComponent, ReactText } from 'react';

interface IndicatorStatDataProps {
  note?: ReactText;
  value?: ReactText;
}

const IndicatorStatDataViewer: FunctionComponent<IndicatorStatDataProps> = ({ value, note }) => {
  return (
    <p className="spotlight__stat-data">
      {value}
      {note ? <span className="spotlight__stat-data__note">{note}</span> : null}
    </p>
  );
};

export { IndicatorStatDataViewer };
