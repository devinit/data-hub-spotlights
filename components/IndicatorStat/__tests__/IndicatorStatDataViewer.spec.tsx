/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { IndicatorStatDataViewer } from '../IndicatorStatDataViewer';

describe('IndicatorStatDataViewer', () => {
  test('renders default state correctly', () => {
    const renderer = TestRenderer.create(<IndicatorStatDataViewer />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders a value correctly', () => {
    const renderer = TestRenderer.create(<IndicatorStatDataViewer value="My Value" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders a note correctly', () => {
    const renderer = TestRenderer.create(<IndicatorStatDataViewer value="My Note" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders a value and a note correctly', () => {
    const renderer = TestRenderer.create(
      <IndicatorStatDataViewer value="My Value" note={{ content: 'My Note' }} />
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
