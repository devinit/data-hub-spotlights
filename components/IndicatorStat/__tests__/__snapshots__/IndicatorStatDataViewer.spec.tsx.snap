// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndicatorStatDataViewer renders a note correctly 1`] = `
<p
  className="spotlight__stat-data"
>
  My Note
</p>
`;

exports[`IndicatorStatDataViewer renders a value and a note correctly 1`] = `
<p
  className="spotlight__stat-data"
>
  My Value
  <span
    className="spotlight__stat-data__note"
  >
    My Note
  </span>
</p>
`;

exports[`IndicatorStatDataViewer renders a value correctly 1`] = `
<p
  className="spotlight__stat-data"
>
  My Value
</p>
`;

exports[`IndicatorStatDataViewer renders default state correctly 1`] = `
<p
  className="spotlight__stat-data"
/>
`;
